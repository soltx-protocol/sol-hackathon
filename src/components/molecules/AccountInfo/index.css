.account-info {
	width: 100%;

	& > .header {
		font-weight: 700;
		font-size: 24px;
		line-height: 24px;
		color: #d9dcff;
	}

	& > .account {
		font-weight: 400;
		font-size: 14px;
		line-height: 20px;
		color: #d9dcff;
	}

	& > .coin {
		margin-top: 46px;
		font-weight: 400;
		font-size: 14px;
		line-height: 20px;
		color: #d9dcff;
	}
}
