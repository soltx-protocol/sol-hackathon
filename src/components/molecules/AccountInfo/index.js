import React from 'react';
import classnames from 'classnames';

import { useAccount } from 'models/account';
import styles from './index.css';

const AccountInfo = ({ className }) => {
	const [{ account }] = useAccount();

	return (
		<div className={classnames(styles.accountInfo, className)}>
			<div className={styles.header}>Individual Account </div>
			<div className={styles.account}>{account === '' ? '-' : account}</div>
			<div className={styles.coin}>Asset:</div>
		</div>
	);
};

export default AccountInfo;
