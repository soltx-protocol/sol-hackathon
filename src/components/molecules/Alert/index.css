.alert {
	display: flex;
	justify-items: flex-start;
	padding: 16px;
	height: 78px;
	width: 384px;
	border-radius: 8px;
	background-color: var(--neutral-white);
	box-shadow: 0 10px 15px -3px rgb(0 0 0 / 10%), 0 4px 6px -2px rgb(0 0 0 / 5%);

	& > .content {
		display: flex;
		flex-direction: column;
		margin-left: 12px;

		& > .title {
			font-size: 14px;
			font-weight: 500;
			line-height: 20px;
			height: 20px;
			text-align: left;
			color: var(--neutral-control-layer-action);
			margin-bottom: 4px;
		}

		& > span {
			height: 20px;
			font-size: 14px;
			font-weight: 400;
			line-height: 20px;
			text-align: left;
			color: var(--neutral-control-layer);
		}
	}
}
