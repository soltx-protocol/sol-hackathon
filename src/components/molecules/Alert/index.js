import React from 'react';
import classnames from 'classnames';

import { useAlert, ALERT_TYPE } from 'models/alert';

import NormalIcon from 'images/icon/status-success.inline.svg';
// import WarningIcon from 'images/icon/status-warning.inline.svg';
import ErrorIcon from 'images/icon/status-error.inline.svg';

import Modal from '../Modal';

import styles from './index.css';

export const AlertComponent = ({ className, type, message }) => (
	<div className={classnames(styles.alert, className)}>
		{type === ALERT_TYPE.NORMAL && <NormalIcon />}
		{type === ALERT_TYPE.ERROR && <ErrorIcon />}
		<div className={styles.content}>
			<div className={styles.title}>{type === ALERT_TYPE.NORMAL ? 'Success' : 'Error'}</div>
			<span>{message}</span>
		</div>
	</div>
);

const Alert = ({ className }) => {
	const [alert] = useAlert();

	return (
		<Modal className={styles.modal} active={alert.active}>
			<AlertComponent className={className} type={alert.type} message={alert.data} />
		</Modal>
	);
};

export default Alert;
