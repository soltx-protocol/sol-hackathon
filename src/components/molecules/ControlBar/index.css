.control-bar {
	display: flex;
	align-items: center;
	justify-content: space-between;
	height: 90px;
	background: rgb(68 72 131 / 60%);
	padding: 0 36px;

	& > .king {
		display: flex;
		align-items: center;

		& > .description {
			font-style: normal;
			font-weight: 400;
			font-size: 16px;
			line-height: 20px;
			margin-left: 10px;
			white-space: pre-line;
			color: #f8f8ff;
		}
	}

	& > .wallet {
		display: flex;
		align-items: center;

		& > .input {
			width: 500px;
			margin-right: 20px;
		}
	}
}
