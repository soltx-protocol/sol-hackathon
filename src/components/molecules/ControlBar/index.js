import React from 'react';
import classnames from 'classnames';

import Input from 'components/atoms/Input';
import Button from 'components/atoms/Button';

import KingIcon from 'images/icon/king.inline.svg';

import styles from './index.css';

const ControlBar = ({ className }) => (
	<div className={classnames(styles.controlBar, className)}>
		<div className={styles.king}>
			<KingIcon />
			<div className={styles.description}>{'Top Asset Owner\nToday'}</div>
		</div>
		<div className={styles.wallet}>
			<Input className={styles.input} placeholder="Insert wallet address" />
			<Button>Generate</Button>
		</div>
	</div>
);

export default ControlBar;
