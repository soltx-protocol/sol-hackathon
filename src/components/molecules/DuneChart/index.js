import React from 'react';
import classnames from 'classnames';

import styles from './index.css';

export const DuneDailyFeeByPool = ({
	className,
	feeAccount = 'DLWewB12jzGn4wXJmFCddWDeof1Ma4cZYNRv9CP5hTvX',
	poolTokenMint = '3e1W6Aqcbuk2DfHUwRiRcyzpyYRRjg6yhZZcyEARydUX',
}) => (
	<iframe
		className={classnames(styles.duneChart, className)}
		src={`https://dune.com/embeds/630914/1176681/3d573057-a39b-4bdf-9f63-abf84db2dd9f?feeAccount=${feeAccount}&poolTokenMint=${poolTokenMint}`}
		title="Top 5 Traders"
	/>
);

export const DuneTop5Traders = ({ className }) => (
	<iframe
		className={classnames(styles.duneChart, className)}
		src="https://dune.com/embeds/630617/1176454/f4732340-aafa-43e7-984a-6efe22f8d9f4"
		title="Top 5 Traders"
	/>
);

const DuneChart = ({
	className,
	title,
	poolAccount = 'EGZ7tiLeH62TPV1gL8WwbXGzEPa9zmcpVnnkPKKnrE2U',
}) => (
	<iframe
		className={classnames(styles.duneChart, className)}
		src={`https://dune.com/embeds/626001/1167393/2d14ec26-c13f-483d-8093-aedc1c69ab18?pool_account=${poolAccount}`}
		title={title}
	/>
);

export default DuneChart;
