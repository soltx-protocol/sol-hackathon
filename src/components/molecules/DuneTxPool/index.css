.dune-tx-pool {
	width: 500px;
	height: 300px;
}
