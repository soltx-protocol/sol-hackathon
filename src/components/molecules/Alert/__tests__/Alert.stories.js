import React from 'react';
import { storiesOf } from '@storybook/react';
import { select } from '@storybook/addon-knobs/react';

import { AlertComponent } from 'components/molecules/Alert';

import { ALERT_TYPE } from 'models/alert';

const stories = storiesOf('molecules/Alert', module);

stories.add('__interactive', () => (
	<div style={{ padding: '20px' }}>
		<AlertComponent
			type={select('type', Object.keys(ALERT_TYPE), ALERT_TYPE.NORMAL)}
			message="Your order has been filled"
		/>
	</div>
));
