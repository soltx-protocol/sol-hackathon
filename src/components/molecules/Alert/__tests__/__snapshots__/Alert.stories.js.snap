// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    style="padding: 20px;"
  >
    <div
      class="alert"
    >
      <test-file-stub />
      <div
        class="content"
      >
        <div
          class="title"
        >
          Success
        </div>
        <span>
          Your order has been filled
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
